src/cunumeric/matrix/solve.h

```diff
@@ -0,0 +1,34 @@
+/* Copyright 2021-2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include "cunumeric/cunumeric.h"
+
+namespace cunumeric {
+
+class SolveTask : public CuNumericTask<SolveTask> {
+ public:
+  static const int TASK_ID = CUNUMERIC_SOLVE;
+
+ public:
+  static void cpu_variant(legate::TaskContext& context);
+#ifdef LEGATE_USE_CUDA
```

Comment:
any reason that you don't implement an openmp variant? it should be almost the same as the cpu variant except the `openblas_set_num_threads` call.

Reply:
added OpenMP variant.